qiskit/visualization/pulse/matplotlib.py

```diff
@@ -444,23 +494,50 @@ def _draw_table(self, figure, channels, dt, n_valid_waveform):
 
         return ax
 
-    def _draw_snapshots(self, ax, snapshot_channels, dt, y0):
+    @staticmethod
+    def _draw_snapshots(ax: 'matplotlib.axes.Axes',
+                        snapshot_channels: Dict['Channel', EventsOutputChannels],
+                        dt: float, y0: float) -> NoReturn:
+        """Draw snapshots to given mpl axis.
+
+        Args:
+            ax: axis object to draw snapshots.
+            snapshot_channels: Event table of snapshots.
+            dt: Time interval.
+            y0: vertical position to draw the snapshots.
+        """
         for events in snapshot_channels.values():
             snapshots = events.snapshots
             if snapshots:
                 for time in snapshots:
                     ax.annotate(s=u"\u25D8", xy=(time*dt, y0), xytext=(time*dt, y0+0.08),
                                 arrowprops={'arrowstyle': 'wedge'}, ha='center')
 
-    def _draw_framechanges(self, ax, fcs, dt, y0):
-        framechanges_present = True
+    def _draw_framechanges(self, ax: 'matplotlib.axes.Axes',
+                           fcs: Dict[int, 'FrameChangeInstruction'],
+                           dt: float, y0: float) -> NoReturn:
```

Comment:
```suggestion
                           dt: float, y0: float) -> None:
```

Interesting though, I'd never seen this before so I looked it up: https://docs.python.org/3/library/typing.html#typing.NoReturn

It's for functions that always error and never return.